### PR TITLE
Fix Interpreter/dynamic_self.swift test for ExistentialSpecializer

### PR DESCRIPTION
**Pull request by rajbarik**

### CODE REVIEW

**Comment by rajbarik** [comment]
@swift-ci test.

**Comment by rajbarik** [comment]
@swift-ci test linux

**Review by atrick — APPROVED** [review]
I think you should comment why the attribute is necessary, at least in the commit message but preferably in test itself--the optimizer may specialize the call and attempt to inline, but inlining that would defeat the purpose of the test which is intended to exercise IRGen on these function bodies. Inlining may also cause `self` to be destroyed before the print statement executes.

**Comment by slavapestov** [comment]
@rajbarik Does the behavior of the test change without the `@inline(never)`? If so it's a bug in the specializer.

**Comment by slavapestov** [comment]
@atrick I don't understand what you're suggesting here. Optimizations should never change the behavior of any code. The test is run with both optimizations off and on (check-swift-validation and check-swift-validation-optimize) and the latter has caught optimizer bugs before.

**Comment by slavapestov** [comment]
Oh I see. How about we just remove the print statements from the deinits.

**Comment by atrick** [comment]
@slavapestov I think it's a problem in general when IRGen unit tests are written in .swift. I agree it's best not to change the RUN line. So the question is whether you still want the test to be useful at -O. My point is that *if* the test is actually useful, then those print statements will be in the correct order. It seems strange to me to write passing CHECKs for a test that's already broken. But I don't have a strong opinion.

**Comment by slavapestov** [comment]
@atrick The part of the test that tests when the deinit occurs is not very useful. We should remove it. The rest is still useful. It's mostly an execution test for code generated by IRGen, but it has randomly turned up optimizer crashes in the past, mostly because our coverage of DyamicSelfType elsewhere in the optimizer wasn't great.

**Comment by atrick** [comment]
@slavapestov sounds fine to me. I was just guessing at the purpose of the test, and couldn't help but notice that IRGen will never even see a `dynamicself` in the -O build.

@rajbarik I guess you can just remove the print statements in the deinitializers, then you don't need the `@inline(never)`.

**Comment by rajbarik** [comment]
@swift-ci test.

**Comment by rajbarik** [comment]
@swift-ci test linux

File: test/Interpreter/dynamic_self.swift [hunk 1]
```diff
@@ -47,7 +48,6 @@ class C : P, CP {
   }
 
   deinit {
```
Comment (slavapestov) [inline test/Interpreter/dynamic_self.swift]: I would suggest removing the deinit altogether

File: test/Interpreter/dynamic_self.swift [hunk 2]
```diff
@@ -30,6 +30,7 @@ func callDynamicSelfClassExistential(_ cp: CP) {
 }
 
 struct S : P {
+
```
Comment (slavapestov) [inline test/Interpreter/dynamic_self.swift]: Please remove the blank line

**Comment by rajbarik** [comment]
@swift-ci test.

**Pull request merged**